--- a/src/graphics.c
+++ b/src/graphics.c
@@ -29,13 +29,13 @@
 const u16 gBattleAnimSpritePal_EnergyBall[] = INCGFX_U16("graphics/battle_anims/sprites/energy_ball.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_MegaStone[] = INCGFX_U32("graphics/battle_anims/sprites/mega_stone.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_MegaStone[] = INCGFX_U16("graphics/battle_anims/sprites/mega_stone.png", ".gbapal");
+const u16 gBattleAnimSpritePal_MegaStone[16] = INCGFX_U16("graphics/battle_anims/sprites/mega_stone.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_MegaParticles[] = INCGFX_U32("graphics/battle_anims/sprites/mega_particles.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_MegaParticles[] = INCGFX_U16("graphics/battle_anims/sprites/mega_particles.png", ".gbapal");
+const u16 gBattleAnimSpritePal_MegaParticles[16] = INCGFX_U16("graphics/battle_anims/sprites/mega_particles.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_MegaSymbol[] = INCGFX_U32("graphics/battle_anims/sprites/mega_symbol.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_MegaSymbol[] = INCGFX_U16("graphics/battle_anims/sprites/mega_symbol.png", ".gbapal");
+const u16 gBattleAnimSpritePal_MegaSymbol[16] = INCGFX_U16("graphics/battle_anims/sprites/mega_symbol.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_AlphaSymbol[] = INCGFX_U32("graphics/battle_anims/sprites/alpha_symbol.png", ".4bpp.smol");
 const u16 gBattleAnimSpritePal_AlphaSymbol[] = INCGFX_U16("graphics/battle_anims/sprites/alpha_symbol.png", ".gbapal");
@@ -383,19 +383,19 @@
 const u32 gBattleAnimSpriteGfx_WavingHand[] = INCGFX_U32("graphics/battle_anims/sprites/waving_hand.png", ".4bpp.smol");

 const u32 gBattleAnimSpriteGfx_Feint[] = INCGFX_U32("graphics/battle_anims/sprites/feint_punch.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_Feint[] = INCGFX_U16("graphics/battle_anims/sprites/feint_punch.png", ".gbapal");
+const u16 gBattleAnimSpritePal_Feint[16] = INCGFX_U16("graphics/battle_anims/sprites/feint_punch.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_TrumpCard[] = INCGFX_U32("graphics/battle_anims/sprites/trump_cards.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_TrumpCard[] = INCGFX_U16("graphics/battle_anims/sprites/trump_cards.png", ".gbapal");
+const u16 gBattleAnimSpritePal_TrumpCard[16] = INCGFX_U16("graphics/battle_anims/sprites/trump_cards.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_TrumpCardParticles[] = INCGFX_U32("graphics/battle_anims/sprites/trump_card_particles.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_TrumpCardParticles[] = INCGFX_U16("graphics/battle_anims/sprites/trump_card_particles.png", ".gbapal");
+const u16 gBattleAnimSpritePal_TrumpCardParticles[16] = INCGFX_U16("graphics/battle_anims/sprites/trump_card_particles.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_Acupressure[] = INCGFX_U32("graphics/battle_anims/sprites/acupressure.png", ".4bpp.smol");
 const u16 gBattleAnimSpritePal_Acupressure[] = INCGFX_U16("graphics/battle_anims/sprites/acupressure.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_WringOut[] = INCGFX_U32("graphics/battle_anims/sprites/wring_out.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_WringOut[] = INCGFX_U16("graphics/battle_anims/sprites/wring_out.png", ".gbapal");
+const u16 gBattleAnimSpritePal_WringOut[16] = INCGFX_U16("graphics/battle_anims/sprites/wring_out.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_ClosingEye[] = INCGFX_U32("graphics/battle_anims/sprites/closing_eye.png", ".4bpp.smol");
 const u16 gBattleAnimSpritePal_ClosingEye[] = INCGFX_U16("graphics/battle_anims/sprites/closing_eye.png", ".gbapal");
@@ -439,7 +439,7 @@
 const u16 gBattleAnimSpritePal_WorrySeed[] = INCGFX_U16("graphics/battle_anims/sprites/worry_seed.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_SmallCloud[] = INCGFX_U32("graphics/battle_anims/sprites/small_cloud.png", ".4bpp.smol");
-const u16 gBattleAnimSpritePal_SmallCloud[] = INCGFX_U16("graphics/battle_anims/sprites/small_cloud.png", ".gbapal");
+const u16 gBattleAnimSpritePal_SmallCloud[16] = INCGFX_U16("graphics/battle_anims/sprites/small_cloud.png", ".gbapal");

 const u32 gBattleAnimSpriteGfx_AttackOrder[] = INCGFX_U32("graphics/battle_anims/sprites/attack_order.png", ".4bpp.smol");
 const u16 gBattleAnimSpritePal_AttackOrder[] = INCGFX_U16("graphics/battle_anims/sprites/attack_order.png", ".gbapal");
