--- a/src/script_menu.c
+++ b/src/script_menu.c
@@ -382,7 +382,6 @@
     u8 newWidth;
     u8 taskId;
     u32 windowHeight;
-    struct ListMenu *list;
 
     for (i = 0; i < argc; ++i)
     {
@@ -420,8 +419,7 @@
     gTasks[taskId].data[5] = argc;
     gTasks[taskId].data[7] = maxBeforeScroll;
     StoreWordInTwoHalfwords((u16*) &gTasks[taskId].data[3], (u32) items);
-    list = (void *) gTasks[gTasks[taskId].data[0]].data;
-    ListMenuChangeSelectionFull(list, TRUE, FALSE, initialRow, TRUE);
+    ListMenuChangeSelectionFullByTask(gTasks[taskId].data[0], TRUE, FALSE, initialRow, TRUE);
 
     if (sDynamicMenuEventId != DYN_MULTICHOICE_CB_NONE && sDynamicListMenuEventCollections[sDynamicMenuEventId].OnSelectionChanged)
     {
@@ -540,6 +538,7 @@
     if (done)
     {
         struct ListMenuItem *items;
+        u32 itemsAddress;
 
         PlaySE(SE_SELECT);
 
@@ -556,7 +555,8 @@
             RemoveScrollIndicatorArrowPair(gTasks[taskId].data[6]);
         }
 
-        LoadWordFromTwoHalfwords((u16*) &gTasks[taskId].data[3], (u32* )(&items));
+        LoadWordFromTwoHalfwords((u16*) &gTasks[taskId].data[3], &itemsAddress);
+        items = (struct ListMenuItem *)(uintptr_t)itemsAddress;
         FreeListMenuItems(items, gTasks[taskId].data[5]);
         TRY_FREE_AND_SET_NULL(sDynamicMenuEventScratchPad);
         DestroyListMenuTask(gTasks[taskId].data[0], NULL, NULL);
